src/commcare_cloud/ansible/roles/upgrade_warning/tasks/main.yml

```diff
@@ -0,0 +1,34 @@
+---
+- name: Gather fact 'distribution_version'
+  setup:
+    gather_subset:
+      - '!all'
+      - '!min'
+      - 'distribution_version'
+
+- name: Create groups based on 'distribution_version'
+  ansible.builtin.group_by:
+    key: ubuntu_{{ ansible_distribution_version | replace(".", "_") }}
+
+- name: Show 'distribution_version' for non 22.04 hosts
+  debug:
+    msg: "{{ item }}"
+  register: warning_present
+  loop: "{% if 'ubuntu_22_04' not in group_names %}{{ warning_host_message }}{% else %}[]{% endif %}"
```

Comment:
Based on our discussion, I removed the confusing and unnecessary variable and simplified the loop, but I'd like to keep it (instead of using `when:`) so that the output will not include the skipped hosts.
I reference the generated list in the warning message, and I think it would be helpful to keep it more lean/readable.